src/v/cluster/rm_stm.cc

```diff
@@ -911,8 +914,31 @@ ss::future<result<kafka_result>> rm_stm::do_idempotent_replicate(
   model::record_batch_reader br,
   raft::replicate_options opts,
   ss::lw_shared_ptr<available_promise<>> enqueued,
-  ssx::semaphore_units& units) {
-    auto request = producer->try_emplace_request(bid, synced_term);
+  ssx::semaphore_units& units,
+  bool unknown_producer) {
```

Comment:
is `unknown_producer` indicating that it was the first time the `rm_stm` 'saw' the producer_id ?

Reply:
yep clarified with a bool_struct.